solver/src/settlement_submission.rs

```diff
@@ -31,13 +33,22 @@ pub async fn submit(
     gas_price_cap: f64,
     settlement: Settlement,
 ) -> Result<()> {
-    let nonce = transaction_count(contract)
-        .await
-        .context("failed to get transaction_count")?;
     let settlement = encode_settlement(&settlement)?;
     // Check that a simulation of the transaction works before submitting it.
     simulate_settlement(&settlement, contract).await?;
 
+    let nonce = transaction_count(contract)
+        .await
+        .context("failed to get transaction_count")?;
+    let address = match &contract.defaults().from {
```

Comment:
nanonit: why not expect?

Reply:
I would also slightly prefer expect here.

Reply:
The reason was that this is a reference but unwrap/exect consume so can't use them. Changed it to clone + expect.